src/aap_eda/services/ruleset/activate_rulesets.py

```diff
@@ -191,6 +191,14 @@ def activate(self, activation: models.Activation) -> None:
                 instance,
                 activation_db_logger,
             )
+            activation_db_logger.flush()
```

Comment:
The activation instance may be removed by deactivation when this line is called. It will raise exception and not update the following status.

Reply:
Doing the `activation.save(`) in `_log_activate_failure` would address this?  Even if not, having one branch in `_log_activate_failure` doing a save and the other not doesn't seem correct.

Reply:
Deactivation does not remove any instance or activation. If the activation/instance were deleted then we don't need to update the status.

Reply:
The activation/instance may be deleted during activation is running. Activation itself doesn't know anything about deletion, so will continue to update the status.

Reply:
I have further enhanced the error handling. If we catch error from a deleted DB record, just log the error and stop.